classes/config_settings.php

```diff
@@ -0,0 +1,259 @@
+     *
+     * @throws \dml_exception
+     * @throws \invalid_parameter_exception
+     * @throws \moodle_exception
+     */
+    public static function with_form_data(\stdClass $quiz) {
+        $settings = new \stdClass;
+        if (empty($quiz->id)) {
+            throw new \invalid_parameter_exception('Cannot construct config_settings without $quiz->id');
+        }
+
+        foreach ($quiz as $key => $quizsetting) {
+            if (strpos($key, 'seb_') === 0) {
+                $settings->$key = $quizsetting;
+            }
+        }
+
+        return new self($quiz->id, $settings);
+    }
+
+    /**
+     * Check if the settings exist.
+     *
+     * @return bool Whether any new or existing settings were found for quiz.
+     */
+    public function exists() {
+        // Cast settings to array as empty stdClass will return true.
+        return !empty((array) $this->settings);
+    }
+
+    /**
+     * Get the generic object containing all quiz settings.
+     *
+     * @return \stdClass An object containing all new and existing settings for quiz.
+     */
+    public function get_settings() : \stdClass {
+        return $this->settings;
+    }
+
+    /**
+     * Get the itemid of the upoaded seb config file if it exists.
+     *
+     * @return string Item ID of stored file.
+     */
+    public function get_config_itemid() {
+        if ($this->exists() && !empty($this->settings->sebconfigfile)) {
+            return $this->settings->sebconfigfile;
+        }
+        return '';
+    }
+
+    /**
+     * Save the settings to the DB. May be an update or a create.
+     *
+     * @return bool Whether or not the settings could be saved.
+     */
+    public function save_settings() : bool {
+        global $DB;
+        $success = false;
+
+        // If existing settings, update record, otherwise create record.
+        try {
+            if ($this->existingsettingsid !== -1) {
+                $this->settings->id = $this->existingsettingsid; // Set the existing id of the settings.
+                $success = $DB->update_record(self::SETTINGS_TABLE, $this->settings);
+            } else {
+                $this->existingsettingsid = $DB->insert_record(self::SETTINGS_TABLE, $this->settings);
+                // Check if the insert was successful, and if not, reset existingsettingsid to -1.
+                if ($this->existingsettingsid === false) {
+                    $this->existingsettingsid = -1;
+                } else {
+                    $success = true;
+                }
+            }
+        } catch (\dml_exception $e) {
+            debugging('quizaccess_seb - save settings: ' . $e->getMessage());
+        }
+
+        return $success;
+    }
+
+    /**
+     * Delete files and settings from DB for this quiz.
+     *
+     * @return bool Whether or not settings were deleted successfully.
+     *
+     * @throws \coding_exception
+     * @throws \dml_exception
+     */
+    public function delete_settings() : bool {
```

Comment:
looking at the names of methods... as long as we are in the class already, it feels like we can drop "_settings" postfix/ E.g. delete_settings() -> delete()